### PR TITLE
Rekor Entries should be reconstructed and compared

### PR DESCRIPTION
**Pull request by loosebazooka**
Recreates entry from materials and compares it to the provided log entry

### CODE REVIEW

**Comment by loosebazooka** [comment]
actually I think we should just recreate the entry and compare it.

File: sigstore-java/src/main/java/dev/sigstore/KeylessVerifier.java [hunk 1]
```diff
@@ -182,6 +186,23 @@ public void verify(byte[] artifactDigest, Bundle bundle, VerificationOptions opt
       throw new KeylessVerificationException("Rekor entry signature was not valid", ex);
     }
 
+    // verify the log entry is relevant to the provided verification materials
+    try {
+      var calculatedHashedRekord =
+          Base64.toBase64String(
+              HashedRekordRequest.newHashedRekordRequest(
```
[the remark this answers is not in the document]
Comment (haydentherapper) [reply sigstore-java/src/main/java/dev/sigstore/KeylessVerifier.java:193]: Answered my question, this is the canonicalized form.

File: sigstore-java/src/main/java/dev/sigstore/KeylessVerifier.java [hunk 2]
```diff
@@ -182,6 +186,23 @@ public void verify(byte[] artifactDigest, Bundle bundle, VerificationOptions opt
       throw new KeylessVerificationException("Rekor entry signature was not valid", ex);
     }
 
+    // verify the log entry is relevant to the provided verification materials
+    try {
+      var calculatedHashedRekord =
```
[the remark this answers is not in the document]
Comment (haydentherapper) [reply sigstore-java/src/main/java/dev/sigstore/KeylessVerifier.java:191]: Seems like DSSE isn't implemented yet based on open PRs?

**Review by haydentherapper — APPROVED** [review]

File: sigstore-java/src/main/java/dev/sigstore/KeylessVerifier.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (loosebazooka) [reply sigstore-java/src/main/java/dev/sigstore/KeylessVerifier.java:191]: yeah no dsse yet.

File: sigstore-java/src/main/java/dev/sigstore/KeylessVerifier.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (loosebazooka) [reply sigstore-java/src/main/java/dev/sigstore/KeylessVerifier.java:193]: Yeah this was originally designed to calculate a UUID in the log to get a GET from rekor (pre-bundle).

**Pull request merged**